appinventor/blocklyeditor/src/blocks/components.js

```diff
@@ -405,7 +427,17 @@ Blockly.Blocks.component_event = {
   },
   // Return a list of parameter names
   getParameters: function () {
-    /** @type {EventDescriptor} */
+   /** @type {EventDescriptor} */
+   var defaultParameters = this.getDefaultParameters_();
+   var explicitParameterNames = this.getExplicitParameterNames_();
+   var params = [];
+   for (var i = 0; i < defaultParameters.length; i++) {
+     var paramName = explicitParameterNames[i] || defaultParameters[i].name;
+     params.push({name: paramName, type: defaultParameters[i].type});
+   }
+   return params;
```

Comment:
Indentation looks like it's only +1 here rather than +2.

Reply:
Fixed.